pkg/template_generator.go

```diff
@@ -130,6 +130,19 @@ func generateTemplateForPodSpec(podSpec kapi.PodSpec, key string, value map[stri
 		podSpec.RestartPolicy = kapi.RestartPolicy(fmt.Sprintf("{{.Values.%s.%s}}", key, RestartPolicy))
 	}
 
+	if len(podSpec.ImagePullSecrets) != 0 {
+		imagePullSecretsObj := []kapi.LocalObjectReference{}
+
+		for _, imagePullSecrets := range podSpec.ImagePullSecrets {
+			imagePullSecrets.Name = generateSafeKey(imagePullSecrets.Name)
+			value["imagePullSecrets"] = imagePullSecrets.Name
```

Comment:
So this means I cannot use Chartify to generate Charts for Secrets even when separated from deployment. This is because even in separate Charts as well, every time i deploy Secret Charts, the name will change as per "{{ template "fullname" . }}-secret" & each time i'll need to modify my deployment Values.yml for secret name. This doesn't seems correct atleast for secrets.

For other resources i can agree with multiple deployments in same namespace, but maintaining & updating secret name will become cumbersome with every new deployment. 
Even in example you sent above, the secret name is in sync with deployment.yml as they are part of same Chart. 
But naming doesn't suffice when installing Secrets from different Charts. Atleast for Secret naming should be constant as it is been referenced by other resources.